src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go

```diff
@@ -35,6 +36,15 @@ type OutputRoomEvent struct {
 	db                 *storage.SyncServerDatabase
 	notifier           *sync.Notifier
 	query              api.RoomserverQueryAPI
+	serverName         gomatrixserverlib.ServerName
+	keyID              gomatrixserverlib.KeyID
+	privateKey         []byte
+}
+
+type prevMembership struct {
+	PrevContent events.MemberContent `json:"prev_content"`
```

Comment:
( `json.RawMessage` is a named type for a `[]byte` containing raw JSON which implements `json.Marshaller` by returning the raw JSON.)